pl-csb-model/train/model_finetuner.py

```diff
@@ -9,73 +10,91 @@
 
 from configparser import ConfigParser
 
-def cleanup() -> None:
-    """Try to free GPU memory"""
-    gc.collect()
-    torch.cuda.empty_cache()
+class ModelFinetuner:
+    __logger: Logger
 
-def get_random_language_pairs(batch_size, langs, data):
-    (l1, long1), (l2, long2) = random.sample(langs, 2)
-    xx, yy = [], []
-    for _ in range(batch_size):
-        item = data.iloc[random.randint(0, len(data)-1)]
-        xx.append(item[l1])
-        yy.append(item[l2])
-    return xx, yy, long1, long2
+    def __init__(self, logger) -> None:
+        self.__logger = logger
 
-def train(model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, data: pd.DataFrame, optimizer: Adafactor, config: ConfigParser) -> None:
-    train_conf = config["TRAINING"]
+    def __cleanup(self) -> None:
```

Comment:
Can `collect` or `empty_cache` throw any exceptions? If so, catch them.

Reply:
No, neither gc.collect() nor torch.cuda.empty_cache() are likely to throw exceptions under normal circumstances.